compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3750,7 +3751,7 @@ public void visit(BLangRawTemplateLiteral rawTemplateLiteral) {
     private BType determineRawTemplateLiteralType(BLangRawTemplateLiteral rawTemplateLiteral, BType expType) {
         // Contextually expected type is NoType when `var` is used.
         // Therefore consider the literal as of type RawTemplate
-        if (expType == symTable.noType || expType == symTable.anyType) {
+        if (expType == symTable.noType || containsAnyType(expType)) {
             return symTable.rawTemplateType;
         }
 
```

Comment:
Btw, the following union as the CET should also work right?

```ballerina
type Foo abstract object {
    public string[] strings;
    public any[] insertions;
};

public function main() {
    string x = "World";
    Foo|int f = `"Hello ${x}`;
}
```

Currently (slp1) fails

```cmd
$ ballerina run qwe.bal 
Compiling source
        qwe.bal
error: .::qwe.bal:8:17: invalid literal for type '(Foo|int)': raw templates can only be assigned to abstract subtypes of 'ballerina/lang.object:1.0.0:RawTemplate'
```

Reply:
And the following union, where only one is compatible.

```ballerina
type Foo abstract object {
    public string[] strings;
    public int[] insertions;
};

type Bar abstract object {
    public string[] strings;
    public string[] insertions;
};

public function main() {
    string x = "World";
    Foo|Bar f = `"Hello ${x}`;
}
```

Fails with
```cmd
$ ballerina run qwe.bal 
Compiling source
        qwe.bal
error: .::qwe.bal:13:17: invalid raw template assignment: '(Foo|Bar)' should be an abstract object
```

Reply:
Fixed the first case with ce3ad42. The second case is a bit tricky. Looks like the whole type checking logic for raw templates will have to be refactored to support it. I think for now better to defer this.

Reply:
Shall we create an issue for this then? And should we consider giving a different error for now, something along the lines of "not yet supported" maybe?

Reply:
Even to give an error like that, I think we'll have to separately identify that particular case, which is the issue with the way currently it's type checked.

Reply:
Created #24766 to track this